### PR TITLE
(GH-1689) Remove packaging script before upgrade

### PR DESCRIPTION
**Pull request by gep13**
It is possible, due to the way that Package Reducer works, that there
are extra files left in the package directory when doing an upgrade
of a package.  This can cause a package upgrade to fail.  Prior to
performing the upgrade, delete all packaging scripts (chocolateyInstall,
chocolateyUninstall, and chocolateyBeforeModify) from the package
folder, these will then be replaced during the upgrade, if they are
still required.

### CODE REVIEW

File: src/chocolatey/infrastructure.app/services/NugetService.cs [hunk 1]
```diff
@@ -1167,6 +1169,26 @@ public virtual void backup_existing_version(ChocolateyConfiguration config, IPac
             }
         }
 
+        public virtual void remove_packaging_files_prior_to_upgrade(string directoryPath)
+        {
+            // Due to the way that Package Reducer works, there is a potential that a Chocolatey Packaging
+            // script could be incorrectly replaced during an upgrade operation.  To guard against this,
```
Comment (ferventcoder) [inline src/chocolatey/infrastructure.app/services/NugetService.cs]: incorrectly left in place, not replaced. We want it replaced. 👍

**Comment by ferventcoder** [comment]
As expected, there is an integration test that verifies what is removed and what should not be, so it will need adjusted to account for this. I can manage grabbing that if I finish it tonight. Otherwise you may want to review the failed AppVeyor log and take a look at the test to update it for this behavior.

**Comment by gep13** [comment]
@ferventcoder I went down the route of changing the tests, but then I realised that the function that I added wasn't constrained to only happen during an upgrade, which I believe it should be.  I have added this in, and the tests that were failing are now all passing again.

Let me know if you want additional changes.

**Comment by ferventcoder** [comment]
This was rebased against stable and merged in at https://github.com/chocolatey/choco/commit/cb2799b09f0a9c91e652e264e2dfc496c9724323. Thanks!

**Pull request closed without merging**